tests/acceptance/features/component/steps/context_update.py

```diff
@@ -39,37 +39,85 @@
+
+
+@step(u'the context notification has these context elements')
+def the_context_notification_has_these_context_elements(context):
+
+    # Prepare table data
+    context.context_elements = dict()
+    for element in context.table.rows:
+        data = element.as_dict()
+        data = dataset_utils.generate_fixed_length_params(data)
+        data = dataset_utils.remove_missing_params(data)
+        context.context_elements.update(data)
+
+
 @step(u'a context notification is received for "(?P<server_id>.*)" with values')
 def a_context_update_is_received(context, server_id):
 
-    testdata = dataset_utils.prepare_data(context.table)
+    # Prepare table data
+    testdata = dict()
+    for element in context.table.rows:
+        auxdata = element.as_dict()
+        auxdata = dataset_utils.generate_fixed_length_params(auxdata)
+        auxdata = dataset_utils.remove_missing_params(auxdata)
+        testdata.update(auxdata)
+
     attribute_list = list()
-    for header in context.table.headings:
-        attribute_list.append({"name": header, "type": "string", "value": testdata.rows[0][header]})
+    for data in testdata:
+        attribute_list.append({"name": data, "type": "string", "value": testdata[data]})
 
+    type = context.context_elements['type'] if 'type' in context.context_elements else None
+    is_pattern = context.context_elements['isPattern'] if 'isPattern' in context.context_elements else None
+    id = context.context_elements['id'] if 'id' in context.context_elements else None
 
+    print("> Send a context notification to FIWARE-FACTS.")
     context.response = context.facts_client.send_monitored_data(subscription_id = str(uuid.uuid1()),
                                                                 originator=server_id,
                                                                 status_code="200",
-                                                                details="OK", type="vm",
-                                                                is_pattern="false",
-                                                                id=server_id,
+                                                                details="OK",
+                                                                type=type,
+                                                                is_pattern=is_pattern,
+                                                                id=id,
                                                                 attribute_list=attribute_list,
                                                                 tenant_id=context.tenant_id_facts,
                                                                 server_id=server_id)
 
+
 @step(u'the context is updated')
 def the_context_is_updated(context):
 
     assert_that(context.response.ok,
                 "Response to CB notification is not the expected one: Message: {}".format(context.response.text))
 
+
+@step(u'the HTTP "(?P<error_code>.*)" is returned')
```

Comment:
I suggest to change the variable error_code to other name due to you take 200 in some cases. Why not status_code?